EssentialsDiscord/src/com/earth2me/essentials/discord/EssentialsJDA.java

```diff
@@ -0,0 +1,111 @@
+package com.earth2me.essentials.discord;
+
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.JDABuilder;
+import net.dv8tion.jda.api.entities.Activity;
+import net.dv8tion.jda.api.entities.Emote;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.entities.Role;
+import net.dv8tion.jda.api.entities.TextChannel;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.event.HandlerList;
+
+import javax.security.auth.login.LoginException;
+import java.util.Objects;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+
+import static com.earth2me.essentials.I18n.tl;
+
+public class EssentialsJDA {
+
+    private static final Logger logger = Logger.getLogger("EssentialsDiscord");
+
+    private JDA jda = null;
+    private Guild guild;
+    private final EssentialsDiscord plugin;
+
+    public EssentialsJDA(EssentialsDiscord plugin) {
+        this.plugin = plugin;
+    }
+
+    public void startup() throws LoginException, InterruptedException {
+        if (jda != null) {
+            jda.removeEventListener(jda.getRegisteredListeners());
+            HandlerList.unregisterAll(plugin);
+        }
+
+        logger.log(Level.INFO, "Attempting to login to discord...");
+        this.jda = JDABuilder.createDefault(plugin.getSettings().getBotToken())
+                .addEventListeners(new DiscordListener(plugin))
+                .setAutoReconnect(true)
+                .setContextEnabled(false)
+                .build()
+                //We don't want to be async here since players could *possibly* chat before jda starts
+                .awaitReady();
+        this.setActivity(plugin.getSettings().getStatusActivity(), plugin.getSettings().getStatusMessage());
+        logger.log(Level.INFO, "Successfully logged in as " + jda.getSelfUser().getAsTag());
+        try {
+            this.guild = Objects.requireNonNull(this.jda.getGuildById(plugin.getSettings().getGuildId()));
+        } catch (NullPointerException e) {
+            logger.severe("No guild configured!");
+            Bukkit.getPluginManager().disablePlugin(plugin);
+            return;
+        }
+        Bukkit.getServer().getPluginManager().registerEvents(new BukkitListener(plugin, this), plugin);
+    }
+
+    public void sendMessage(String message) {
+        sendMessage(plugin.getSettings().getPrimaryChannelId(), message);
+    }
+
+    public void sendMessage(long channelId, String message) {
+        TextChannel channel = this.jda.getTextChannelById(channelId);
+        if (channel == null) {
+            logger.log(Level.WARNING, tl("discordInvalidChannelId"));
+            return;
+        }
+        String msg = ChatColor.stripColor(message)
+                .replaceAll("@here", "@\u200bhere")
+                .replaceAll("@everyone", "@\u200beveryone");
```

Comment:
That does no longer work from what I remember, as discord just removes zero width space now.
You could however set a `MessageAction.setDefaultMentions(EnumSet.of(Message.MentionType.USER, Message.MentionType.ROLE))` before starting JDA to set the new default mentions, which would automatically disable everyone/here mentions in the process.

More Info:
https://ci.dv8tion.net/job/JDA/javadoc/net/dv8tion/jda/api/requests/restaction/MessageAction.html#setDefaultMentions(java.util.Collection)

Reply:
I attempted to use this, but it seems the API is new and it wasn't working properly at the time I tested it. The zero width space seemed to work, however (but perhaps it's better to simply remove the mention altogether).